karton/core/backend/base.py

```diff
@@ -0,0 +1,92 @@
+import dataclasses
+import enum
+from collections import namedtuple
+from typing import IO, Any, Iterator, Protocol
+
+from karton.core.task import Task, TaskState
+
+KartonBind = namedtuple(
+    "KartonBind",
+    [
+        "identity",
+        "info",
+        "version",
+        "persistent",
+        "filters",
+        "service_version",
+        "is_async",
+    ],
+)
+
+
+class KartonMetrics(enum.Enum):
+    TASK_PRODUCED = "karton.metrics.produced"
+    TASK_CONSUMED = "karton.metrics.consumed"
+    TASK_CRASHED = "karton.metrics.crashed"
+    TASK_ASSIGNED = "karton.metrics.assigned"
+    TASK_GARBAGE_COLLECTED = "karton.metrics.garbage-collected"
+
+
+@dataclasses.dataclass(frozen=True)
+class KartonServiceInfo:
+    """
+    Extended Karton service information.
+
+    Instances of this dataclass are meant to be aggregated to count service replicas
+    in Karton Dashboard. They're considered equal if identity and versions strings
+    are the same.
+    """
+
+    identity: str
+    karton_version: str
+    service_version: str | None = None
+
+
+class KartonBackendProtocol(Protocol):
+    """
+    Interface for KartonBackend high-level methods used by producers and consumers
```

Comment:
Maybe explain a bit more why this exists:


```suggestion
    Protocol that defines methods that KartonBackend must implement.

    Used by producers and consumers to avoid depending on a concrete implementation.
```